crates/cairo-profiler/src/trace_reader/function_trace_builder.rs

```diff
@@ -131,11 +143,53 @@ pub fn collect_function_level_profiling_info(
 
         match gen_statement {
             GenStatement::Invocation(invocation) => {
-                if matches!(
-                    sierra_program_registry.get_libfunc(&invocation.libfunc_id),
-                    Ok(CoreConcreteLibfunc::FunctionCall(_))
-                ) {
-                    call_stack.enter_function_call(current_call_stack);
+                match sierra_program_registry.get_libfunc(&invocation.libfunc_id) {
+                    Ok(CoreConcreteLibfunc::FunctionCall(_)) => {
+                        call_stack.enter_function_call(current_call_stack);
+                    }
+                    Ok(CoreConcreteLibfunc::StarkNet(libfunc)) => {
+                        let syscall = match libfunc {
+                            StarkNetConcreteLibfunc::CallContract(_)
+                            | StarkNetConcreteLibfunc::Deploy(_)
+                            | StarkNetConcreteLibfunc::EmitEvent(_)
+                            | StarkNetConcreteLibfunc::GetBlockHash(_)
+                            | StarkNetConcreteLibfunc::GetExecutionInfo(_)
+                            | StarkNetConcreteLibfunc::GetExecutionInfoV2(_)
+                            | StarkNetConcreteLibfunc::Keccak(_)
+                            | StarkNetConcreteLibfunc::LibraryCall(_)
+                            | StarkNetConcreteLibfunc::ReplaceClass(_)
+                            | StarkNetConcreteLibfunc::SendMessageToL1(_)
+                            | StarkNetConcreteLibfunc::StorageRead(_)
+                            | StarkNetConcreteLibfunc::StorageWrite(_)
+                            | StarkNetConcreteLibfunc::Sha256ProcessBlock(_) => libfunc,
+                            _ => {
+                                if in_syscall_idx.is_some() {
+                                    in_syscall_idx = None;
+                                }
```

Comment:
[nit] could just be `in_syscall_idx = None`

Reply:
clippy then complains about incosistent formatting (https://rust-lang.github.io/rust-clippy/master/index.html#semicolon_if_nothing_returned) :(

Reply:
No, i mean the `if` is redundant

Reply:
yup, so this was already done in https://github.com/software-mansion/cairo-profiler/pull/112/commits/263b43c4cc27240828856471e3fe9b6ddc5933ce